utils/oomparser/oomparser.go

```diff
@@ -101,132 +90,46 @@ func checkIfStartOfOomMessages(line string) bool {
-	lineChannel := make(chan string, 10)
-	go func() {
-		readLinesFromFile(lineChannel, self.ioreader)
-	}()
+// StreamOoms writes to a provided a stream of OomInstance objects representing
+// OOM events that are found in the logs.
+// It will blocks and should be called from a goroutine.
+func (self *OomParser) StreamOoms(outStream chan<- *OomInstance) {
+	kmsgEntries := self.parser.Parse()
+	defer self.parser.Close()
 
-	for line := range lineChannel {
-		in_oom_kernel_log := checkIfStartOfOomMessages(line)
+	for msg := range kmsgEntries {
+		in_oom_kernel_log := checkIfStartOfOomMessages(msg.Message)
 		if in_oom_kernel_log {
 			oomCurrentInstance := &OomInstance{
 				ContainerName: "/",
+				TimeOfDeath:   msg.Timestamp,
 			}
-			for line := range lineChannel {
-				err := getContainerName(line, oomCurrentInstance)
+			for msg := range kmsgEntries {
+				err := getContainerName(msg.Message, oomCurrentInstance)
 				if err != nil {
 					glog.Errorf("%v", err)
 				}
-				finished, err := getProcessNamePid(line, oomCurrentInstance)
+				finished, err := getProcessNamePid(msg.Message, oomCurrentInstance)
 				if err != nil {
 					glog.Errorf("%v", err)
 				}
 				if finished {
+					oomCurrentInstance.TimeOfDeath = msg.Timestamp
 					break
 				}
 			}
 			outStream <- oomCurrentInstance
 		}
 	}
-	glog.Infof("exiting analyzeLines. OOM events will not be reported.")
-}
-
-func callJournalctl() (io.ReadCloser, error) {
-	cmd := exec.Command("journalctl", "-k", "-f")
-	readcloser, err := cmd.StdoutPipe()
-	if err != nil {
-		return nil, err
-	}
-	if err := cmd.Start(); err != nil {
-		return nil, err
-	}
-	return readcloser, err
-}
-
-func trySystemd() (*OomParser, error) {
-	readcloser, err := callJournalctl()
-	if err != nil {
-		return nil, err
-	}
-	glog.Infof("oomparser using systemd")
-	return &OomParser{
-		ioreader: bufio.NewReader(readcloser),
-	}, nil
-}
-
-// List of possible kernel log files. These are prioritized in order so that
-// we will use the first one that is available.
-var kernelLogFiles = []string{"/var/log/kern.log", "/var/log/messages", "/var/log/syslog"}
-
-// looks for system files that contain kernel messages and if one is found, sets
-// the systemFile attribute of the OomParser object
-func getLogFile() (string, error) {
-	for _, logFile := range kernelLogFiles {
-		if utils.FileExists(logFile) {
-			glog.Infof("OOM parser using kernel log file: %q", logFile)
-			return logFile, nil
-		}
-	}
-	return "", fmt.Errorf("unable to find any kernel log file available from our set: %v", kernelLogFiles)
-}
-
-func tryLogFile() (*OomParser, error) {
-	logFile, err := getLogFile()
-	if err != nil {
-		return nil, err
-	}
-	tail, err := tail.NewTail(logFile)
-	if err != nil {
-		return nil, err
-	}
-	return &OomParser{
-		ioreader: bufio.NewReader(tail),
-	}, nil
+	// Should not happen
+	glog.Warningf("exiting analyzeLines. OOM events will not be reported.")
```

Comment:
Before it was info and I thought it should be error, so I picked the middle as a compromise :innocent: